src/domain/challenge/challenge.module.ts

```diff
@@ -3,10 +3,36 @@ import { UserGroupModule } from '../user-group/user-group.module';
 import { UserGroupService } from '../user-group/user-group.service';
 import { ChallengeService } from './challenge.service';
 import { ChallengeResolver } from './challenge.resolver';
+import { UserModule } from '../user/user.module';
+import { UserService } from '../user/user.service';
+import { ContextModule } from '../context/context.module';
+import { ContextService } from '../context/context.service';
+import { ReferenceModule } from '../reference/reference.module';
+import { ReferenceService } from '../reference/reference.service';
+import { TagsetModule } from '../tagset/tagset.module';
+import { TagsetService } from '../tagset/tagset.service';
+import { ProfileModule } from '../profile/profile.module';
+import { ProfileService } from '../profile/profile.service';
 
 @Module({
-  providers: [ChallengeService, UserGroupService, ChallengeResolver],
-  imports: [UserGroupModule],
+  imports: [
+    ContextModule,
+    ProfileModule,
+    ReferenceModule,
+    TagsetModule,
+    UserGroupModule,
+    UserModule,
+  ],
+  providers: [
```

Comment:
I only see UserGroupService, ContextService and TagsetService injected in ChallengeService. All the unnecessary references need to be removed - no need to bloat the dependency graph.

Reply:
I had tried that - but it seems dependencies pass through in Nest. If they were not there it gave errors. My conclusion was that modules do *not* encapsulate.